### ISSUE TITLE
Builds error when using next's i18n due to lack of `serverless/pages/index.html` file

### ISSUE THREAD

**Issue by neefrehman** [netlify/netlify-plugin-nextjs#75]
I'm trying to move a Next.js site (which uses next's i18n routing) over to Netlify, but sadly can't due to this error:

```
Error: ENOENT: no such file or directory, stat '.next/serverless/pages/index.html'
```

It's true that this file doesn't exist. Instead, the default homepage can be found at  '.next/serverless/pages/en/index.html' (replacing "en" with whatever default locale is set in next.config.js).

It would be great to detect if i18n is in use, and then get the homepage's file location by reading the default locale.

**Comment by cantyjeffrey** [comment]
Same issue here

**Comment by lindsaylevine** [comment]
hey yall! @neefrehman @cantyjeffrey our i18n PR in next-on-netlify was merged and released this past sunday https://github.com/netlify/next-on-netlify/pull/75. i will go ahead and upgrade the plugin's next-on-netlify version now. admittedly, we need a better workflow for keeping the plugin up to date with next-on-netlify updates. thank you for your patience!!!

**Comment by lindsaylevine** [comment]
update: i upgraded the next-on-netlify version and re-released the plugin. i18n should work now in plugin version 1.0.4!

**Comment by neefrehman** [comment]
@lindsaylevine thanks forthe update!

I've just tried deploying a sit with the new plugin and am getting a similar error. It looks like the plugin is searching for the entry point at `pages/${locale}/.html` as opposed to `pages/${locale}/index.html`

```bash
10:52:20 AM: ────────────────────────────────────────────────────────────────
10:52:20 AM:   Plugin "@netlify/plugin-nextjs" internal error                
10:52:20 AM: ────────────────────────────────────────────────────────────────
10:52:20 AM: ​
10:52:20 AM:   Error message
10:52:20 AM:   Error: ENOENT: no such file or directory, stat '.next/serverless/pages/en/.html'
10:52:20 AM: ​
10:52:20 AM:   Plugin details
10:52:20 AM:   Package:        @netlify/plugin-nextjs
10:52:20 AM:   Version:        1.0.4
10:52:20 AM:   Repository:     git+https://github.com/netlify/netlify-plugin-nextjs.git
10:52:20 AM:   npm link:       https://www.npmjs.com/package/@netlify/plugin-nextjs
10:52:20 AM:   Report issues:  https://github.com/netlify/netlify-plugin-nextjs/issues
```

**Comment by lindsaylevine** [comment]
oh no what page type is your index!? @neefrehman thats so odd after my thorough testing 🤔

**Comment by lindsaylevine** [comment]
actually thats really not odd i had opened this issue for myself 😭 https://github.com/netlify/next-on-netlify/issues/122

**Comment by neefrehman** [comment]
hah! well if it's still useful my index page is just a standard `index.tsx` in my pages directory

**Comment by lindsaylevine** [comment]
im able to repro, ill keep ya updated. apologies!

**Comment by lindsaylevine** [comment]
i believe this is the fix https://github.com/netlify/next-on-netlify/pull/131 now just need to go through the re-release of everything 😅

btw when i asked for page type (sorry that was unclear) i meant are you using getStaticProps (likely since the build is looking for html) and are you using fallback and/or revalidate

**Comment by neefrehman** [comment]
Ah understood! Yes it's a `getStaticProps` page without any revalidation or fallback :)

**Comment by lindsaylevine** [comment]
@neefrehman ok 1.0.5 is released now. hopefully you have a clean build and functioning i18n with this version! 🙏

**Comment by neefrehman** [comment]
I've actually just tried a build! It's moving past the old error, but now I'm getting the below error regarding function bundling.

I guess the important piece is : `Cannot find module 'critters'`

I'm not sure if this is related?

```bash
11:35:56 AM: ────────────────────────────────────────────────────────────────
11:35:56 AM:   Dependencies installation error                               
11:35:56 AM: ────────────────────────────────────────────────────────────────
11:35:56 AM: ​
11:35:56 AM:   Error message
11:35:56 AM:   A Netlify Function failed to require one of its dependencies.
11:35:56 AM:   If the dependency is a Node module, please make sure it is present in the site's top-level "package.json".
  If it is a local file instead, please make sure the file exists and its filename is correctly spelled.
11:35:56 AM: ​
11:35:56 AM:   In file "/opt/build/repo/netlify-automatic-functions/next_index/next_index.js"
11:35:56 AM:   Cannot find module 'critters'
11:35:56 AM:   Require stack:
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/node_modules/@netlify/zip-it-and-ship-it/src/node_dependencies/resolve.js
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/node_modules/@netlify/zip-it-and-ship-it/src/node_dependencies/index.js
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/node_modules/@netlify/zip-it-and-ship-it/src/main.js
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/src/plugins_core/functions/index.js
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/src/commands/get.js
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/src/core/main.js
11:35:56 AM:   - /opt/buildhome/.netlify-build-nvm/versions/node/v12.16.3/lib/node_modules/@netlify/build/src/core/bin.js
11:35:56 AM: ​
11:35:56 AM:   Resolved config
11:35:56 AM:   build:
11:35:56 AM:     command: npm run build
11:35:56 AM:     commandOrigin: config
11:35:56 AM:     environment:
11:35:56 AM:       - NODE_VERSION
11:35:56 AM:     publish: /opt/build/repo/out
11:35:56 AM:   plugins:
11:35:56 AM:     - inputs: {}
11:35:56 AM:       origin: config
11:35:56 AM:       package: netlify-plugin-cache-nextjs
11:35:56 AM:     - inputs: {}
11:35:56 AM:       origin: config
11:35:56 AM:       package: '@netlify/plugin-nextjs'
```

**Comment by lindsaylevine** [comment]
oh dear. this is a known issue as well! see: https://twitter.com/erikras/status/1341783510055567361?s=20 and https://github.com/netlify/next-on-netlify/issues/103

temporary fix is to install `critters` in your project. question though before doing that - can you try adding a next.config.js (if you dont already have one) and setting the target to `experimental-serverless-trace` (if it isn't already set to that)?

**Comment by lindsaylevine** [comment]
also, best to reply on 103 if you can! regarding critters. if i18n is all good for you here, i'll close this issue :)

**Comment by neefrehman** [comment]
@lindsaylevine sorry for opening another issue again, but I've just updated the plugin to fix the resurfaced `critters` issue (https://github.com/netlify/next-on-netlify/issues/157), and now and getting one similar to this again. It looks like the plugin is looking for a file like `pages/{defaultLocale}/{defaultLocale}.html` instead of `pages/{defaultLocale}/index.html`.

More info:
```bash
11:34:25 AM: ────────────────────────────────────────────────────────────────
11:34:25 AM:   Plugin "@netlify/plugin-nextjs" internal error                
11:34:25 AM: ────────────────────────────────────────────────────────────────
11:34:25 AM: ​
11:34:25 AM:   Error message
11:34:25 AM:   Error: ENOENT: no such file or directory, stat '.next/serverless/pages/en/en.html'
11:34:25 AM: ​
11:34:25 AM:   Plugin details
11:34:25 AM:   Package:        @netlify/plugin-nextjs
11:34:25 AM:   Version:        1.1.0
11:34:25 AM:   Repository:     git+https://github.com/netlify/netlify-plugin-nextjs.git
11:34:25 AM:   npm link:       https://www.npmjs.com/package/@netlify/plugin-nextjs
11:34:25 AM:   Report issues:  https://github.com/netlify/netlify-plugin-nextjs/issues

**Comment by lindsaylevine** [comment]
@neefrehman don't be sorry at all!!! yikes, that's not good though @ bug. let me try to repro.

**Comment by lindsaylevine** [comment]
repro'd, stay tuned

**Comment by lindsaylevine** [comment]
this issue is haunting me https://github.com/netlify/next-on-netlify/issues/122

**Comment by lindsaylevine** [comment]
i assume you're using next 10.0.6 because something definitely changed internally :rip:

**Comment by lindsaylevine** [comment]
aight, i know what's going on. are you blocked by this? if so, i think the workaround is to downgrade next until i can release a fix

**Comment by neefrehman** [comment]
great! i'm not blocked, so happy to downgrade for now. it must be a big challenge to continue covering all cases

**Comment by lindsaylevine** [comment]
this was just released! let me know if you continue to have any issues!
